src/schema/quo.cljs

```diff
@@ -2,27 +2,75 @@
   (:require
     [schema.registry :as registry]))
 
-(def ^:private ?profile-picture-fn-params
+(def ^:private ?profile-picture-options
   [:map
    [:length :int]
    [:full-name :string]
    [:font-size :int]
    [:indicator-size {:optional true} [:maybe :int]]
    [:indicator-color {:optional true} [:maybe :string]]
    [:indicator-center-to-edge {:optional true} [:maybe :int]]
-   [:override-theme :schema.common/theme]
+   [:theme {:optional true} [:maybe :schema.common/theme]]
    [:background-color :string]
    [:color :string]
    [:size :int]
    [:ring? :boolean]
    [:ring-width :int]])
 
+(def ^:private ?account-image-uri-options
+  [:map
+   [:port :int]
+   [:ratio :double]
+   [:key-uid :string]
+   [:image-name :string]
+   [:theme :schema.common/theme]
+   [:override-ring? [:maybe :boolean]]])
+
+(def ^:private ?initials-image-uri-options
+  [:map
+   [:port :int]
+   [:ratio :double]
+   [:uppercase-ratio :double]
+   [:font-file :string]
+   [:theme :schema.common/theme]
+   [:key-uid {:optional true} [:maybe :string]]
+   [:public-key {:optional true} [:maybe :string]]
+   [:override-ring? {:optional true} [:maybe :boolean]]])
+
+(def ^:private ?contact-image-uri-options
+  [:map
+   [:port :int]
+   [:clock :int]
+   [:ratio :double]
+   [:image-name :string]
+   [:public-key :string]
+   [:theme :schema.common/theme]
+   [:override-ring? [:maybe :boolean]]])
+
+(def ^:private ?image-uri-config
+  [:or
+   [:map
+    [:kind [:enum :contact]]
+    [:options ?contact-image-uri-options]]
+   [:map
+    [:kind [:enum :account]]
+    [:options ?account-image-uri-options]]
+   [:map
+    [:kind [:enum :initials]]
+    [:options ?initials-image-uri-options]]])
+
 (def ^:private ?profile-picture-source
   [:or
    :schema.common/image-source
    [:map
-    [:fn [:=> [:cat ?profile-picture-fn-params] :string]]]])
+    [:config ?image-uri-config]]])
 
 (defn register-schemas
   []
-  (registry/register ::profile-picture-source ?profile-picture-source))
+  (registry/register ::profile-picture-options ?profile-picture-options)
+  (registry/register ::image-uri-config ?image-uri-config)
+  (registry/register ::profile-picture-source ?profile-picture-source)
+  (registry/register ::profile-picture-options ?profile-picture-options)
+  (registry/register ::account-image-uri-options ?account-image-uri-options)
+  (registry/register ::contact-image-uri-options ?contact-image-uri-options)
+  (registry/register ::initials-image-uri-options ?initials-image-uri-options))
```

Comment:
do we need all these individual schemas? If I understand correctly we would only use the `get-image-uri` helper and pass around the :kind/:type. In this case it seems only `?image-uri-config` would have to be exposed.

Reply:
`:multi` most likely will help with error reporting :+1:

Reply:
One thing that's interesting is that using `:or` with `:enum` results with an error message that will say:
```
 {:in [0 :type],
   :message "should be :initials",
   :path [0 2 :type],
   :schema [:enum :initials],
   :value :contact}
```
when passing the wrong `:type` value (`:contact`) but with the correct options for `:initials`. So I guess this could be an extra hint, but I'm not sure how useful it is. Otherwise, using `:multi` provides clear reporting about how to change the options to match the provided `:type`. Any preference?

Reply:
no strong preference on this really, just that I guess :type doesn't have to be an enum, since it can be only one value for each variant. You can use `[:type [:= :initials]]`.

Reply:
nice! I'll switch to using that 🙌